http/src/main/scala/filodb/http/HealthRoute.scala

```diff
@@ -27,6 +25,21 @@ class HealthRoute(coordinatorActor: ActorRef, v2ClusterEnabled: Boolean, setting
                                          ShardStatusRecovery.getClass,
                                          ShardStatusAssigned.getClass)
 
+  def checkIfAllShardsLiveClusterV1(shardEvents: collection.Map[DatasetRef, Seq[ShardEvent]]): Boolean = {
+    shardEvents.values.flatten.forall {
+      case IngestionStarted(_, _, _) | RecoveryStarted(_, _, _, _) | RecoveryInProgress(_, _, _, _) => true
+      case _ => false
+    }
+  }
+
+  def checkIfAllShardsLiveClusterV2(shardHealthSeq: Seq[DatasetShardHealth]): Boolean = {
```

Comment:
If there are `N` shards in the cluster, is it possible that the V2 coordinator node returns fewer than `N` statuses? I.e. if a single host is unresponsive?

Reply:
This ask call on the coordinator queries only the given instance/host which is queried. It will not coordinate with other hosts. Hence, we will only get the local shard health information. 

Saying that, I have added additional check to make sure that all the local shards responses is collected before we make a decision for liveness.